blockly/src/utils/pathfinding/BFSPathFinding.java

```diff
@@ -0,0 +1,50 @@
+package utils.pathfinding;
+
+import core.level.utils.Coordinate;
+import java.util.*;
+
+/**
+ * Breadth-First Search (BFS) pathfinding algorithm implementation.
+ *
+ * <p>This class implements the PathfindingLogic interface and provides the logic for performing BFS
+ * pathfinding. It maintains the open and closed sets, tracks the path, and provides methods to
+ * initialize the search, perform steps, and retrieve the final path.
```

Comment:
Das diese Klasse die `PathfindingLogic` implementiert, sehe ich bereits am `extends` 😝

Mich würde hier viel mehr interessieren, was Du hier mit open und closed sets betreibst, da man in der Regel nur die Arbeits-Queue plus ggf. bei Graphsearch die "closed set" hat. Also der Algorithmus in Pseudocode.